### ISSUE TITLE
Testing Enum with flags gives error

### ISSUE THREAD

**Issue by tl24** [hazzik/DelegateDecompiler#87]
If I have a flags enum like this:

```
[Flags]
public enum MyFlags {
    Flag1 = 1 << 1,
    Flag2 = 1 << 2
}
```

and have a property defined like this:

```
public class MyEntity {
     public MyFlags MyFlagsProperty {get; set; }

     [Computed]
     public bool Flag1IsSet {
           get { return (MyFlagsProperty & MyFlags.Flag1) != 0; }
     }
}
```

Trying to access Flag1IsSet property in a query results in this error:
The binary operator And is not defined for the types 'MyFlags ' and 'System.Int32'

Using MyFlagsProperty.IsSet(MyFlags.Flag1) does work and is probably the better way of doing it, but the decompiler is doing some sort of conversion on the other format that it probably shouldn't be doing.

**Comment by hazzik** [comment]
Thanks for reporting

**Comment by hazzik** [comment]
Released as 0.23.0
